Dev Guide: No mention of setting `CLUSTER_IDENTITY_NAME` for creating workload cluster

As part of https://capz.sigs.k8s.io/developers/development.html#customizing-the-cluster-deployment , there's no mention of setting `CLUSTER_IDENTITY_NAME` environment variable but it's used in the template yaml - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/621f596ed7a2f46d7c6a6a239a35e9eea7385380/templates/cluster-template.yaml#L211

Due to this, when creating workload cluster using `make create-workload-cluster` command, we get an error

```bash
error: error when retrieving current configuration of:
Resource: "infrastructure.cluster.x-k8s.io/v1alpha4, Resource=azureclusteridentities", GroupVersionKind: "infrastructure.cluster.x-k8s.io/v1alpha4, Kind=AzureClusterIdentity"
Name: "", Namespace: "default"
from server for: "STDIN": resource name may not be empty
```

Full log -

```bash
cluster-api-provider-azure $ make create-workload-cluster
# Create workload Cluster.
/Users/karuppiahn/projects/github.com/kubernetes-sigs/cluster-api-provider-azure/hack/tools/bin/envsubst-drone < /Users/karuppiahn/projects/github.com/kubernetes-sigs/cluster-api-provider-azure/templates/cluster-template.yaml | kubectl apply -f -
cluster.cluster.x-k8s.io/capz-cluster created
azurecluster.infrastructure.cluster.x-k8s.io/capz-cluster created
kubeadmcontrolplane.controlplane.cluster.x-k8s.io/capz-cluster-control-plane created
azuremachinetemplate.infrastructure.cluster.x-k8s.io/capz-cluster-control-plane created
machinedeployment.cluster.x-k8s.io/capz-cluster-md-0 created
azuremachinetemplate.infrastructure.cluster.x-k8s.io/capz-cluster-md-0 created
kubeadmconfigtemplate.bootstrap.cluster.x-k8s.io/capz-cluster-md-0 created
error: error when retrieving current configuration of:
Resource: "infrastructure.cluster.x-k8s.io/v1alpha4, Resource=azureclusteridentities", GroupVersionKind: "infrastructure.cluster.x-k8s.io/v1alpha4, Kind=AzureClusterIdentity"
Name: "", Namespace: "default"
from server for: "STDIN": resource name may not be empty
make: *** [create-workload-cluster] Error 1
```

--- a/docs/book/src/developers/development.md
+++ b/docs/book/src/developers/development.md
@@ -297,7 +297,7 @@ CLUSTER_NAME=<my-capz-cluster-name> ./hack/create-dev-cluster.sh
    2.2. Push to your custom image registry:
 
    ```bash
-   REGISTRY="<container-registry>" MANAGER_IMAGE_TAG="<image-tag>" make docker-push
+   REGISTRY=${REGISTRY} MANAGER_IMAGE_TAG=${MANAGER_IMAGE_TAG:="dev"} make docker-push
    ```
 
    NOTE: `make create-cluster` will fetch the manager image locally and load it onto the kind cluster if it is present.
@@ -325,6 +325,11 @@ export AZURE_CONTROL_PLANE_MACHINE_TYPE="Standard_D2s_v3"
 export AZURE_NODE_MACHINE_TYPE="Standard_D2s_v3"
 export WORKER_MACHINE_COUNT=2
 export KUBERNETES_VERSION="v1.22.1"
+
+# Identity secret.
+export AZURE_CLUSTER_IDENTITY_SECRET_NAME="cluster-identity-secret" 
+export CLUSTER_IDENTITY_NAME="cluster-identity" 
+export AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE="default"
 
 # Generate SSH key.
 # If you want to provide your own key, skip this step and set AZURE_SSH_PUBLIC_KEY_B64 to your existing file.

